whois-api/src/main/java/net/ripe/db/whois/api/rdap/WhoisRdapService.java

```diff
@@ -205,10 +227,44 @@ private Boolean isRedirect(ObjectType objectType, final String key) {
     }
 
     protected Response lookupObject(final HttpServletRequest request, final Set<ObjectType> objectTypes, final String key) {
-        List<RpslObject> result =  rdapQueryHandler.handleQuery(getQueryObject(objectTypes, key), request);
+        Iterable<RpslObject> result =  rdapQueryHandler.handleQuery(getQueryObject(objectTypes, key), request);
         return getResponse(request, result);
     }
 
+    protected Response lookupForOrganisation(final HttpServletRequest request, final Set<ObjectType> objectTypes,
+                                             final String key) {
+        final List<RpslObject> organisationStream = rdapQueryHandler.handleQuery(getQueryObject(objectTypes,
+                        key),
+                request);
+
+        if(organisationStream.isEmpty()){
+            throw new NotFoundException("not found");
+        }
+        if (organisationStream.size() > 1){
+            throw new IllegalStateException("Unexpected result size: " + organisationStream.size());
+        }
+
+        final Query autnumInetnumForOrganisationQuery = Query.parse(
+                String.format("%s %s %s %s %s %s %s %s",
+                        QueryFlag.NO_GROUPING.getLongFlag(),
+                        QueryFlag.NO_REFERENCED.getLongFlag(),
+                        QueryFlag.SELECT_TYPES.getLongFlag(),
+                        objectTypesToString(List.of(AUT_NUM, ObjectType.INETNUM, ObjectType.INET6NUM)),
+                        QueryFlag.NO_FILTERING.getLongFlag(),
+                        QueryFlag.INVERSE.getLongFlag(),
+                        AttributeType.ORG.getName(),
+                        key));
+
+        final Stream.Builder<RpslObject> resourcesStream = rdapQueryHandler.handleQueryIter(autnumInetnumForOrganisationQuery,
```

Comment:
handleQueryIter should be renamed to streamQueryResults or sth like that. btw it's probably better to return a Stream than a Stream.Builder